### ISSUE TITLE
No output with new Zig version

### ISSUE THREAD

**Issue by captainkirkoff** [shilangyu/scoop-search#46]
Hi.
With release 1.4.0 I just have no output when I invoke it.
Any way to investigate the issue? What could I try?
Thanks.

![image](https://github.com/shilangyu/scoop-search/assets/6802846/29ec5ca3-f7b5-47da-a050-7fd1eda3eaa1)

**Comment by shilangyu** [comment]
I am seeing correctly that it exits with a non-zero status code? It must be crashing for some reason.

Could you show me your installed buckets (`scoop bucket list`)?

**Comment by captainkirkoff** [comment]
> scoop bucket list

Here they are:

```
Name                 Source                                        Updated             Manifests
----                 ------                                        -------             ---------
bodrick_scoop-bucket https://github.com/bodrick/scoop-bucket       24/01/2024 21:26:22        21
extras               https://github.com/ScoopInstaller/Extras      02/02/2024 13:53:38      1958
main                 https://github.com/ScoopInstaller/Main        02/02/2024 09:25:16      1287
nonportable          https://github.com/ScoopInstaller/Nonportable 02/02/2024 13:42:10       127

```

**Comment by captainkirkoff** [comment]
And yes, return code is not 0:

![image](https://github.com/shilangyu/scoop-search/assets/6802846/50ab7d2c-a9d9-481e-9347-786a1efc72d3)

**Comment by shilangyu** [comment]
Argh that looks like a segfault. Unfortunately I cannot reproduce this using the same buckets you have, so there must be something specific about your setup. This sucks, as there is not much I can do to debug this. I would probably have to do #47 to give some debug info at runtime.

Just to make sure someone else isn't killing the process. Does `scoop-search --hook` work? Also are you getting no output regardless of the searched term?

**Comment by captainkirkoff** [comment]
> Just to make sure someone else isn't killing the process. Does `scoop-search --hook` work? Also are you getting no output regardless of the searched term?

No, it does not work.
I usually don't use PowerShell, but I tried to activate the hook in $PROFILE and I get:

```
PowerShell 7.4.1
Invoke-Expression: C:\Users\piero\Documents\PowerShell\Microsoft.PowerShell_profile.ps1:5
Line |
   5 |  Invoke-Expression (&scoop-search --hook)
     |                    ~~~~~~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Command' because it is null.
Loading personal and system profiles took 3294ms.
```
It also occurs with built-in Windows 10 Powershell.

I just tried scoop-search with another computer of mine, same buckets, Windows 11 (the faulty one has Win 10) and it works flawlessly, both in cmd and as an hook in powershell.

If you add some verbose/debug option in next release I will be happy to investigate more.

**Comment by captainkirkoff** [comment]
> ➜ scoop-search --hook
> function scoop { if ($args[0] -eq "search") { scoop-search.exe @($args | Select-Object -Skip 1) } else { scoop.ps1 @args } }
> ```

Ok, sorry for not metioning it, but I did try that indeed at first: no output also with --hook in my Windows 10 machine (I do get it in my Windows 11 one).

As regard the antivirus question: I only have Windows Security active and there's no evidence that defender's blocking it. It occurred with past version (1.3.3): scoop-search.exe was quarantined and I whtitelisted it. As well I whitelisted in advance the executable in current folder.

**Comment by RioNight** [comment]
I don't know if this is worth mentioning, but scoop-search exits with a bad exit code when run with no arguments in cmd.
It seemingly hangs for a few seconds and outputs nothing.

![image](https://github.com/shilangyu/scoop-search/assets/58403689/4a73e62b-d9d3-490a-91f6-bbbd326b23f6)

**Comment by shilangyu** [comment]
The above is not related to this issue.

**Comment by captainkirkoff** [comment]
Hi,
today I updated scoop-search to 1.4.1 and I still had the issue.
Then I wanted to do a further try: I installed zig, cloned the repo and built on my own.
I made my local build on the master branch and it works flawlessly, both building it with no parameter (zig build) and with -Doptimize=ReleaseFast.
I even overwritten current scoop scoop-search binary (even freshly reinstalled and not working) with my build, and it works.

![image](https://github.com/shilangyu/scoop-search/assets/6802846/ab0a62b6-c267-455f-8de9-acfd29f9e19b)

I can't imagine why, but it seems the my problem is not with your code but with the github build.

**Comment by shilangyu** [comment]
This aligns with the antivirus theory. You building the binary locally probably makes some AV not scan and block it since it's not foreign. No idea though. None of it makes sense.

**Comment by captainkirkoff** [comment]
> This aligns with the antivirus theory. You building the binary locally probably makes some AV not scan and block it since it's not foreign. No idea though. None of it makes sense.

I have to agree: it totally shows up as an antivirus issue, even though Windows Defender isn't giving me any evidence that scoop-search.exe is a threat, it shows no threat history, the binary location is even actually in whitelist, etc, and the issue persists even disabling the real time protection.

If Windows Defender (the only antivirus I have here) is somehow preventing me to properly launch that binary, I just don't know if and how I can take control.

**Comment by shilangyu** [comment]
Are you able to completely disable windows defender temporarily to see if scoop search won't work still? Another issue that this could be related to is maybe a missing DLL, but windows loves to show popups for that and you said nothing happens :/

**Comment by captainkirkoff** [comment]
> Are you able to completely disable windows defender temporarily to see if scoop search won't work still? Another issue that this could be related to is maybe a missing DLL, but windows loves to show popups for that and you said nothing happens :/

Yes, in theory at least I can disable all Microsoft Defender temporarily and it doesn't help. Also, if I lack a dll, it should affect my own build as well, shouldn't it?
I'll try to investigate a bit more and I'll keep you updated.

**Comment by captainkirkoff** [comment]
Ok, inspecting the event viewer I can confirm that scoop-search.exe (yours) is searching for something it cannot find.
But it's not clear what it's searching for.
The application error has an event id 1005. Normally it specifies the name of the missing file.
All that I can see here is a space character.

![image](https://github.com/shilangyu/scoop-search/assets/6802846/ecbb955b-1825-4236-9e22-9409c77b1dfd)

```
Impossibile accedere al file  per uno dei motivi seguenti: [...]
```

```
[-](https://github.com/shilangyu/scoop-search/issues/46#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/shilangyu/scoop-search/issues/46#) <System>
  <Provider Name="Application Error" /> 
  <EventID Qualifiers="49152">1005</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2024-02-14T09:46:12.6114938Z" /> 
  <EventRecordID>111115</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>S20-PEDRO.pter.loc</Computer> 
  <Security /> 
  </System>
[-](https://github.com/shilangyu/scoop-search/issues/46#) <EventData>
  <Data /> 
  <Data>scoop-search.exe</Data> 
  <Data>00000000</Data> 
  <Data>0</Data> 
  </EventData>
  </Event>
```

**Comment by Sebourg** [comment]
I am running into the same issue. I have the same error (ID 1005) in the event viewer with an empty filename also.

**Comment by captainkirkoff** [comment]
Update.
If I build the executable on my Windows 11 machine (zig and zls installed the same way as in the Windows 10 one, via scoop), the build crashes in the Windows 10 one as the original.
So I tried to debug via WinDbg: it crash after an unmanaged exception (C000001D.ILLEGAL_INSTRUCTION).
Hard enough for me to find where it occurs in the code. I can send a process dump if it can help.

**Comment by shilangyu** [comment]
Just to clarify: you built it on win11 and transferred the binary to a different machine running win10?

Perhaps that is the issue, the binary is built on a different target machine that is for some reason incompatible with some windows versions. The scoop binary is built on windows server 2022. That would be really surprising though, these things should "just work", I would have to dig in zig's compiler options to see if anywhere it makes the assumption about the underlying windows version.

**Comment by captainkirkoff** [comment]
> Just to clarify: you built it on win11 and transferred the binary to a different machine running win10?
> 
> Perhaps that is the issue, the binary is built on a different target machine that is for some reason incompatible with some windows versions. The scoop binary is built on windows server 2022. That would be really surprising though, these things should "just work", I would have to dig in zig's compiler options to see if anywhere it makes the assumption about the underlying windows version.

Yes I did.
I have two machine: the one with Windows 11 has never had issues with scoop-search, the one with Windows 10 was affected by this issue.
When I built scoop-search in the Windows 10 machine it worked (and it works also in the Windows 11 one).
The build made in the Windows 11 machine, instead, behaves like yours built on windows server 2022: it works in Windows 11 and it does not in Windows 10.

**Comment by demosthenez** [comment]
I get the same or similar problems on Windows 11.  scoop-search *anything* and the result is: -1073741795.  It's the same for scoop-search --help or whatever else.

**Comment by sobrinojulian** [comment]
I have the same problem on Windows 10 without antivirus (Defender also disabled). 1.3.3 works fine though

**Comment by hertz98** [comment]
> Update. If I build the executable on my Windows 11 machine (zig and zls installed the same way as in the Windows 10 one, via scoop), the build crashes in the Windows 10 one as the original. So I tried to debug via WinDbg: it crash after an unmanaged exception (C000001D.ILLEGAL_INSTRUCTION). Hard enough for me to find where it occurs in the code. I can send a process dump if it can help.

Same behavior: using a debugger it outputs:
```
EXCEPTION_DEBUG_INFO:
           dwFirstChance: 0
           ExceptionCode: C000001D (EXCEPTION_ILLEGAL_INSTRUCTION)
          ExceptionFlags: 00000000
        ExceptionAddress: scoop-search.[00007FF7D00D2279](x64dbg://localhost/address64#00007FF7D00D2279)
        NumberParameters: 0
Last chance exception on [00007FF7D00D2279](x64dbg://localhost/address64#00007FF7D00D2279) (C000001D, EXCEPTION_ILLEGAL_INSTRUCTION)!
```
And the address is pointing to
```
00007FF7D00D2279 | C462B9F7F8 | shlx r15,rax,r8 | 
```
After searching in the internet, seems that `shlx` instruction belongs to the AVX2 Instruction Set, but unfortunately it is not supported by my old CPU

Not sure about Zig, but to solve the issue it should be compiled with different options in order to disable AVX2 as the following answer says: https://stackoverflow.com/questions/42232218/bit-shift-illegal-instruction-in-64-bit-release-mode-only

**Comment by shilangyu** [comment]
@hertz98 thank you for that, that seems like a good hint. I will have to find the flag that disables AVX2, or check that at runtime. I wonder how this is usually solved, I've never seen projects release separate binaries for various CPU support

**Comment by hertz98** [comment]
In my opinion it's a trade-off between how old are the machines the developer wants to support versus the efficiency of the application (I guess the compiler is using avx2 beacuse the program result faster) 
I don't think is worth two binaries

**Comment by hertz98** [comment]
I did a test on my newer machine, building using zig 0.11, and seems adding `-Dcpu=baseline` flag to the build arguments solve the problem

**Comment by tianlongJ** [comment]
I also encountered the problem that scoop-search did not have any output, version 1.4.1

**Comment by shilangyu** [comment]
Should be fixed in 1.5. Feel free to comment if it is not the case, I will reopen the issue.

**Comment by sobrinojulian** [comment]
Works great for me in my i5-3320M. Thank you!
